packages/react-accordion/Spec.md

```diff
@@ -0,0 +1,369 @@
+
+## Prior art
+
+As a part of the spec definitions in Fluent UI, a research effort has been made through [Open UI](https://open-ui.org/). The current research proposal is available as an open source contribution undergoing review ([research proposal](https://github.com/WICG/open-ui/pull/263))
+
+## Comparison of `@fluentui/react` and `@fluentui/react-northstar`
+
+- All mentions of v7 or v8 == `@fluentui/react` ([docsite](https://developer.microsoft.com/en-us/fluentui#/))
+- All mentions of v0 == `@fluentui/react-northstar` ([docsite](https://fluentsite.z22.web.core.windows.net/))
+
+There's no comparison to be done between the two libraries, since v8 hasn't implemented this component.
+
+## Issues
+
+- [#16553](https://github.com/microsoft/fluentui/issues/16553)
+- [#16452](https://github.com/microsoft/fluentui/issues/16452)
+- [#12953](https://github.com/microsoft/fluentui/issues/12953)
+- [#12238](https://github.com/microsoft/fluentui/issues/12238)
+
+## API
+
+The `Accordion` should implement a `children` based API as is the standard across all the surveyed alternatives as a part of Open UI research in [Prior Art](#prior-art). The component will leverage the use of `context` in the interaction and data flows of child components.
+
+Sample usages will be given in the following section of this document [Sample code](#sample-code)
+
+### Accordion
+
+The root level component serves context and common API between all children.
+
+| Prop name          | Type                | Default Value | Details                                                                                                                    |
+| ------------------ | ------------------- | ------------- | -------------------------------------------------------------------------------------------------------------------------- |
+| multiple           | boolean             | false         | Allows multiple panels to be expanded                                                                                      |
+| collapsible        | boolean             | false         | Allows multiple panels to be collapsed                                                                                     |
+| expandIconPosition | "start" or "end"    | "start"       | Position of the icon to indicate expansion                                                                                 |
+| onToggle           | ToggleEventListener |               | Equivalent to onToggle on AccordionPanel element                                                                           |
+| open               | number or number[]  | undefined     | The index (or a list of indexes) of opened items. If this property is being used than the element is considered Controlled |
```

Comment:
Off-topic:
I don't know it is a common pattern to have virtualized accordions, but in this case it can be an issue if we are using DOM positions as indexes. For example, I will have following markup:

```tsx
<Accordion open={[1]}>
  <AccordionItem>0</AccordionItem>
  <AccordionItem>1</AccordionItem>
  {/* more items */}
</Accordion>
```

On first render a second item will be active:

```html
<div class="accordion">
  <div class="accordion-item">0</div>
  <-- ✅ rendered DOM matches JSX, item is active -->
  <div class="accordion-item active">1</div>
  <div class="accordion-item">2</div>
</div>
```

Once I will scroll a second item **in DOM** will be active:

```html
<div class="accordion">
  <div class="accordion-item">30</div>
  <-- 🟥 rendered DOM is different, a different item is active -->
  <div class="accordion-item active">31</div>
  <div class="accordion-item">32</div>
</div>
```

This might be confusing for consumers. It's a not a big issue now, but we should find a solution for these cases.

Reply:
I don't think virtualization is a common pattern to be used with `Accordion`. To be honest, never seen accordion being used more than a few times in the same page, like 4-5 `AccordionItem`.

If virtualization is something to keep in mind than the [@reach/descendants API](https://github.com/reach/reach-ui/tree/develop/packages/descendants) becomes unviable and we'll have to find another mechanism that doesn't depend on position. Right now I'm assuming that virtualization is a no go for an `AccordionItem`.